pkg/sessions/session_store.go

```diff
@@ -5,17 +5,22 @@ import (
 
 	"github.com/oauth2-proxy/oauth2-proxy/pkg/apis/options"
 	"github.com/oauth2-proxy/oauth2-proxy/pkg/apis/sessions"
+	"github.com/oauth2-proxy/oauth2-proxy/pkg/encryption"
 	"github.com/oauth2-proxy/oauth2-proxy/pkg/sessions/cookie"
 	"github.com/oauth2-proxy/oauth2-proxy/pkg/sessions/redis"
 )
 
 // NewSessionStore creates a SessionStore from the provided configuration
 func NewSessionStore(opts *options.SessionOptions, cookieOpts *options.CookieOptions) (sessions.SessionStore, error) {
+	cipher, err := encryption.NewBase64Cipher(encryption.NewCFBCipher, encryption.SecretBytes(cookieOpts.Secret))
```

Comment:
Does this potentially belong inside of the specific `New<Type>SessionStore` where they can initialize from the `cookieOpts.Secret` they have access to?

One of the things I'm looking at in the session state refactor is more variability of the cipher based on the security scenarios each store has.

At the moment I'm looking at the cookies themselves always using AES-CFB (since it doesn't have repeat IV weakness).  But for the inner second encryption of DB store values, AES-GCM is much better suited.

This works fine with that design (since I intend to use CFB on all store's cookies) -- just was curious if we wanted to future proof ourselves a bit and give the particular session full control of its crypto algorithm decisions.

Reply:
> One of the things I'm looking at in the session state refactor is more variability of the cipher based on the security scenarios each store has.

Hadn't really thought about this as it's not something we've done so far, we've always passed the same cipher through. That said, it makes a lot of sense!

I'll take a look at moving it in, or we could do that in another PR later, up to you

Reply:
Ok I moved it. We need to be vigilant about this and make sure that as we add more session stores that we ensure they're encrypted. I can't remember if the tests cover this or not, but if they don't we should make sure they do (I think there's a test that changes the cookie secret and ensures decryption fails)